### PR TITLE
PostProcessing: Replaced ImageSharpen with ImageFilter processor

### PR DESCRIPTION
**Pull request by PetiteViking**
The ImageFilter processor implements a number of convolution kernels, including the replaced sharpening.
An option would be to incorporate the High/low-pass processors here as well.

### CODE REVIEW

File: modules/basegl/include/modules/basegl/algorithm/imageconvolution.h [hunk 1]
```diff
@@ -49,6 +49,9 @@ class IVW_MODULE_BASEGL_API ImageConvolution {
 
     std::shared_ptr<Image> convolution(const Layer &layer, std::function<float(vec2)> kernelWeight,
                                        const float &kernelScale, ivec2 kernelSize);
+    std::shared_ptr<Image> convolution(const Layer &layer, int kw, int kh,
+                                                const std::vector<float> &kernel,
+                                                const float &kernelScale);
```
Comment (martinfalk) [inline modules/basegl/include/modules/basegl/algorithm/imageconvolution.h]: clang format?

File: modules/postprocessing/CMakeLists.txt [hunk 2]
```diff
@@ -13,9 +13,9 @@ set(HEADER_FILES
     include/modules/postprocessing/processors/hdrbloom.h
     include/modules/postprocessing/processors/imagebrightnesscontrast.h
     include/modules/postprocessing/processors/imageedgedarken.h
+	include/modules/postprocessing/processors/imagefilter.h
```
Comment (martinfalk) [inline modules/postprocessing/CMakeLists.txt]: clang format/use spaces for indentation

**Review by martinfalk — APPROVED** [review]
nice

**Comment by JochenJankowai** [comment]
I don't know how to fix the format error. I reformatted all the files that I have touched using clang but apparently it doesn't help.

**Comment by petersteneteg** [comment]
Seems something goes wrong, I can have a look

**Comment by martinfalk** [comment]
Might this be caused by different clang versions?
```
/Users/jenkins/jenkins/workspace/inviwo_feature_imagefilter/inviwo/modules/basegl/src/algorithm/imageconvolution.cpp (original)
+++ /Users/jenkins/jenkins/workspace/inviwo_feature_imagefilter/inviwo/modules/basegl/src/algorithm/imageconvolution.cpp (formatted)
@@ -69,8 +69,8 @@
 }
 
 std::shared_ptr<Image> ImageConvolution::lowpass(const Layer &layer, int kernelSize) {
-    return convolution_separable(layer, [](float /*p*/) { return 1.f; },
-                                 kernelSize, static_cast<float>(kernelSize));
+    return convolution_separable(layer, [](float /*p*/) { return 1.f; }, kernelSize,
+                                 static_cast<float>(kernelSize));
 }
```

**Comment by r-englund** [comment]
What was the formating problem? UTF encoding?

**Comment by petersteneteg** [comment]
no the file was deleted :) I format all files modified in the pr... and that one was removed :)

**Pull request merged**